account_reconcile_oca/models/account_bank_statement_line.py

```diff
@@ -975,13 +976,22 @@ def create(self, mvals):
             "_test_account_reconcile_oca"
         ):
             return result
-        models = self.env["account.reconcile.model"].search(
-            [
-                ("rule_type", "in", ["invoice_matching", "writeoff_suggestion"]),
-                ("company_id", "in", result.company_id.ids),
-                ("auto_reconcile", "=", True),
+        domain = [
+            ("rule_type", "in", ["invoice_matching", "writeoff_suggestion"]),
+            ("company_id", "in", result.company_id.ids),
+            ("auto_reconcile", "=", True),
+        ]
+        # In case, for example, of bank statement import on a journal
+        journal_id = self.env.context.get("journal_id")
+        if journal_id:
+            domain += [
+                "|",
+                ("match_journal_ids", "=", False),
+                ("match_journal_ids", "in", journal_id),
             ]
```

Comment:
```suggestion
        domain += [
            "|",
            ("match_journal_ids", "=", False),
            ("match_journal_ids", "in", result.journal_id.ids),
        ]
```
Although using the context data should not be a problem, I think it is best to use the journal_id data from the created records.

Reply:
Indeed. I did the change